diff --git a/node_modules/react-native-hce/android/gradlew b/node_modules/react-native-hce/android/gradlew
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-hce/android/src/main/java/com/reactnativehce/services/CardService.java b/node_modules/react-native-hce/android/src/main/java/com/reactnativehce/services/CardService.java
index f666c0f..84ab7ef 100644
--- a/node_modules/react-native-hce/android/src/main/java/com/reactnativehce/services/CardService.java
+++ b/node_modules/react-native-hce/android/src/main/java/com/reactnativehce/services/CardService.java
@@ -27,6 +27,11 @@ public class CardService extends HostApduService {
 
     @Override
     public byte[] processCommandApdu(byte[] command, Bundle extras) {
+      StringBuilder commandHex = new StringBuilder();
+      for (byte b : command) {
+          commandHex.append(String.format("%02X", b));
+      }
+      Log.d(TAG, "Received APDU command: " + commandHex.toString());
       if (currentHCEApplication != null) {
         return currentHCEApplication.processCommand(command);
       }
@@ -45,7 +50,7 @@ public class CardService extends HostApduService {
 
     @Override
     public void onCreate() {
-      Log.d(TAG, "Starting service");
+      Log.d(TAG, "Starting service!!!");
       Context context = getApplicationContext();
 
       registeredHCEApplications.add(new NFCTagType4(
@@ -57,6 +62,8 @@ public class CardService extends HostApduService {
     @Override
     public void onDeactivated(int reason) {
       Log.d(TAG, "Finishing service: " + reason);
-      this.currentHCEApplication.onDestroy(reason);
+      if (this.currentHCEApplication != null) {
+        this.currentHCEApplication.onDestroy(reason);
+      }
     }
 }
